pallets/collator-selection/src/lib.rs

```diff
@@ -406,6 +424,64 @@ pub mod pallet {
 
 			Ok(Some(T::WeightInfo::leave_intent(current_count as u32)).into())
 		}
+
+		/// Add a new account `who` to the list of `Invulnerables` collators.
+		///
+		/// The origin for this call must be the `UpdateOrigin`.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::add_invulnerable(T::MaxInvulnerables::get() - 1))]
+		pub fn add_invulnerable(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			let invulnerables = Self::invulnerables().to_vec();
+
+			// ensure `who` is not already invulnerable
+			ensure!(!invulnerables.contains(&who), Error::<T>::AlreadyInvulnerable);
+
+			// ensure `who` has registred a validator key
+			let validator_key = T::ValidatorIdOf::convert(who.clone())
+				.ok_or(Error::<T>::NoAssociatedValidatorId)?;
+			ensure!(
+				T::ValidatorRegistration::is_registered(&validator_key),
+				Error::<T>::ValidatorNotRegistered
+			);
+
+			<Invulnerables<T>>::try_mutate(|invulnerables| -> DispatchResult {
+				invulnerables
+					.try_push(who.clone())
+					.map_err(|_| Error::<T>::TooManyInvulnerables)?;
+				invulnerables.sort();
```

Comment:
Instead of sorting, could also be implemented like this
```
<Invulnerables<T>>::try_mutate(|invulnerables| -> DispatchResult {
				match invulnerables.binary_search(&who) {
					Ok(_) => return Err(Error::<T>::AlreadyInvulnerable)?,
					Err(pos) => invulnerables
						.try_insert(pos, who.clone())
						.map_err(|_| Error::<T>::TooManyInvulnerables)?,
				}
				Ok(())
			})?;
```

Reply:
Yes, better, thanks